### PR TITLE
feat: Enable parquet predicate pushdown in IOx by default

### PR DESCRIPTION
**Pull request by alamb**
Closes https://github.com/influxdata/influxdb_iox/issues/5887

~Draft as it~ 
- [x] ~builds on https://github.com/influxdata/influxdb_iox/pull/5928~
- [x] upstream https://github.com/apache/arrow-datafusion/pull/3989
- [x] Builds on https://github.com/influxdata/influxdb_iox/pull/6010
- [x] Builds on https://github.com/influxdata/influxdb_iox/pull/6011
- [x] Complete upstream testing https://github.com/apache/arrow-datafusion/pull/3976

# Rationale:
Towards https://github.com/influxdata/idpe/issues/16227

While we have plans to make predicate pushdown even better, we would like to start with it on and then iterate from there. 

See more context on https://github.com/apache/arrow-datafusion/issues/3462 and https://github.com/influxdata/k8s-idpe/issues/11978#issuecomment-1284466216

# Changes:
1. Enable parquet predicate pushdown by default
2. Add tests

# Questions
~I am not sure how to test this~
Update: Testing using parquet metrics

### CODE REVIEW

**Comment by alamb** [comment]
I wonder if we can/should add metrics to the DataFusion parquet reader that shows how many rows (if any) were pruned? If we had that we could write a proper test for this

**Comment by crepererum** [comment]
Yeah, metrics for this would definitely be nice, also for debugging.

One quick win could be to instrument the `SendableRecordBatchStream` that reads from the `ParquetExec` and let this one count rows. However I would kindly ask you to NOT do that because I'm in the process of lifting the `ParquetExec` up so that DataFusion can "see" it (i.e. not hiding it behind some opaque stream).

**Comment by alamb** [comment]
I will look into adding some appropriate metric, etc next week

**Comment by alamb** [comment]
Update here is I am marking this as a draft while I work on some tests / visibility upstream

**Comment by alamb** [comment]
I have a proposed PR to add metrics upstream in DataFusion: https://github.com/apache/arrow-datafusion/pull/3989

**Comment by alamb** [comment]
Updated to test using metrics.

**Comment by alamb** [comment]
Hmm, well I found a few bugs upstream in DataFusion https://github.com/apache/arrow-datafusion/issues/4006 and https://github.com/apache/arrow-datafusion/issues/4005.

I think it needs some more testing before enabling in IOx

File: query_tests/src/sql_metrics.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        );
+
+        // verify that pushdown was enabled and that it filtered rows
+        let metrics = parquet_metrics(physical_plan);
+        assert_eq!(
+            metric_value_sum(&metrics, "pushdown_rows_filtered"),
+            8,
+            "Unexpected number of rows filtered in:\n\n{:#?}",
+            metrics
+        );
```
Comment (alamb) [inline query_tests/src/sql_metrics.rs:72]: This is the actual test showing that rows were filtered

**Review by crepererum — APPROVED** [review]

**Comment by alamb** [comment]
🎉

**Comment by alamb** [comment]
cc @tustvold  I am turning this on

**Pull request merged**